docs/content/config/security/ssl.md

```diff
@@ -478,94 +478,45 @@ DSM-generated letsencrypt certificates get auto-renewed every three months.
-
-```caddyfile
-{
-  debug
-  admin localhost:2019
-  http_port 80
-  https_port 443
-  default_sni example.com
-  key_type rsa4096
-}
-```
-
-If you are instead using a json config for Caddy v2, you can set it in your site's TLS automation policies:
-
-??? example "Caddy v2 JSON example snippet"
-
-    ```json
-    {
-      "apps": {
-        "http": {
-          "servers": {
-            "srv0": {
-              "listen": [
-                ":443"
-              ],
-              "routes": [
-                {
-                  "match": [
-                    {
-                      "host": [
-                        "mail.example.com",
-                      ]
-                    }
-                  ],
-                  "handle": [
-                    {
-                      "handler": "subroute",
-                      "routes": [
-                        {
-                          "handle": [
-                            {
-                              "body": "",
-                              "handler": "static_response"
-                            }
-                          ]
-                        }
-                      ]
-                    }
-                  ],
-                  "terminal": true
-                },
-              ]
-            }
-          }
-        },
-        "tls": {
-          "automation": {
-            "policies": [
-              {
-                "subjects": [
-                  "mail.example.com",
-                ],
-                "key_type": "rsa2048",
-                "issuer": {
-                  "email": "admin@example.com",
-                  "module": "acme"
-                }
-              },
-              {
-                "issuer": {
-                  "email": "admin@example.com",
-                  "module": "acme"
-                }
-              }
-            ]
-          }
-        }
-      }
-    }
-    ```
+[Caddy](https://caddyserver.com/) is an open-source web server with built-in SSL certificate generation. You can use the [official Docker image](https://hub.docker.com/_/caddy) and write your own Caddyfile, or use [lucaslorentz/caddy-docker-proxy](https://github.com/lucaslorentz/caddy-docker-proxy), which lets you add labels to your services in order to generate a Caddyfile.
 
-The generated certificates can then be mounted:
+DMS supports retrieving certificates from Caddy data when the local file system storage is used (the default). You just need to mount your Caddy data and set `SSL_TYPE: letsencrypt` in the DMS container. Here is a minimal example with `lucaslorentz/caddy-docker-proxy`:
 
 ```yaml
+version: "3.7"
+
 volumes:
-  - ${CADDY_DATA_DIR}/certificates/acme-v02.api.letsencrypt.org-directory/mail.example.com/mail.example.com.crt:/etc/letsencrypt/live/mail.example.com/fullchain.pem
-  - ${CADDY_DATA_DIR}/certificates/acme-v02.api.letsencrypt.org-directory/mail.example.com/mail.example.com.key:/etc/letsencrypt/live/mail.example.com/privkey.pem
```

Comment:
And here in the old docs we have an example filepath that your PR presumably operates on and is still roughly the same?

Like with certbot section, it advises mounting the relevant cert folder to `/etc/letsencrypt/live` location that DMS will internally check for with `SSL_TYPE=letsencrypt`:

https://github.com/docker-mailserver/docker-mailserver/blob/8f971713360c4620046c7f223fe66df9ea800253/target/scripts/helpers/ssl.sh#L394-L412

Traefik has special support because of the `acme.json` format. Whereas I feel this PR is adding a very minor convenience to the user?

[Our docs do cover `SSL_TYPE=manual`](https://docker-mailserver.github.io/docker-mailserver/v12.1/config/security/ssl/#bring-your-own-certificates) as an alternative to `SSL_TYPE=letsencrypt`.  Both of which have `check-for-changes.sh` support as their locations are monitored:

https://github.com/docker-mailserver/docker-mailserver/blob/8f971713360c4620046c7f223fe66df9ea800253/target/scripts/helpers/change-detection.sh#L46-L63

However in your current PR state, your approach for Caddy would miss out on this and would not benefit. The Traefik `acme.json` support is hard-coded into our `/etc/letsencrypt` internal location which has been considered acceptable. While for this Caddy support we need to also rely on `/caddy_data` to monitor for changes (_ideally only for those relevant to DMS, which is better supported if mounting the full path to isolate if the reverse-proxy was using the same data folder for other services, as shown with rspamd_).

The difference with normal `SSL_TYPE=letsencrypt` is that unlike `SSL_TYPE=manual` there is no internal copy made. I'd like to simplify the SSL / TLS support, but haven't found time to. Ideally it'd be more generic in finding a cert from some location and then copying it to a fixed internal location. I understand the desire for convenience though.